benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -404,35 +405,37 @@ choice value =
 -------------------------------------------------------------------------------
 
 {-# INLINE parseMany #-}
-parseMany :: MonadThrow m => Int -> Stream m Int -> m ()
+parseMany :: Monad m => Int -> Stream m Int -> m ()
 parseMany n =
       Stream.fold Fold.drain
     . fmap getSum
-    . Stream.parseMany (PR.fromFold $ Fold.take n Fold.mconcat)
+    . Stream.rights . Stream.parseMany (PR.fromFold $ Fold.take n Fold.mconcat)
     . fmap Sum
 
 {-# INLINE parseIterate #-}
-parseIterate :: MonadThrow m => Int -> Stream m Int -> m ()
+parseIterate :: Monad m => Int -> Stream m Int -> m ()
 parseIterate n =
       Stream.fold Fold.drain
     . fmap getSum
-    . Stream.parseIterate (PR.fromFold . Fold.take n . Fold.sconcat) (Sum 0)
+    . Stream.rights
+    . Stream.parseIterate
+        (PR.fromFold . Fold.take n . Fold.sconcat)
+        (Sum 0)
     . fmap Sum
 
 {-# INLINE parseBreak #-}
-parseBreak :: MonadCatch m => Stream m Int -> m ()
+parseBreak :: Monad m => Stream m Int -> m ()
 parseBreak s = do
-    r <- try $ Stream.parseBreak PR.one s
+    r <- Stream.parseBreak PR.one s
     case r of
-        Left (_ :: SomeException) -> return ()
-        Right (_, s1) -> parseBreak s1
+         (_, s1) -> parseBreak s1
```

Comment:
This is causing an infinite loop. We should terminate on Left result as before. I am fixing it.